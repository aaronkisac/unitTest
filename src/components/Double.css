.page {
  margin: 50px;
  text-align: center;
}
.error {
  margin: 50px;
  color: red;
}
label,
input,
button {
  margin: 20px;
  font-size: 20px;
}
