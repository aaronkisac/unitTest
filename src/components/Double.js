import React, { useState } from "react";
import "./Double.css";

const Double = ({ initialNum = 11, calculateDouble }) => {
  
  const [num, setNum] = useState(initialNum);
  const [double, setDouble] = useState(num * 2);
  const [error, setError] = useState("");
  const [disabled, setDisabled] = useState(false);

  const calculate = () => {
    console.log(num, double);
    const newDouble = calculateDouble(num);
    setDouble(newDouble);
  };

  const handleOnChange = (e) => {
    if (e.target.value < 0 || e.target.value > 1000) {
      setError("Please type 0 to 100");
      setDisabled(true);
    } else {
      setNum(e.target.value);
      setDisabled(false);
    }
  };


  return (
    <div id="divDouble" className="page">
      <h5 className="title1"> unit test title1</h5>
      <h5 className="title2"> unit test title2</h5>
      <div className="error">{error && <label>{error}</label>}</div>
      <input
        id="number"
        type="text"
        onChange={handleOnChange}
        defaultValue={num}
      />
      <button onClick={calculate} disabled={disabled}>
        Calculate
      </button>
      <div>
        <label>{double}</label>
      </div>
    </div>
  );
};

export default Double;
