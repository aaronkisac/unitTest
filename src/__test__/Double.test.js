import React from "react";
import { shallow, mount } from "enzyme";
import Double from "../components/Double";

function setup(num=0) {
  const fnc = jest.fn();
  const wrapper = shallow(<Double initialNum={24} calculateDouble={fnc} />);
  return { wrapper, fnc };
}

describe("test", () => {
  it("test not working", () => {
    expect(2).toBe(2);
  });
});

// describe("Double component Cases", () => {
//   it("Check All Attributes", () => {
//     const { wrapper, fnc } = setup();
//     expect(wrapper).toHaveLength(1);
//   });

  
//   //? BEN BURADA CLİCKLEMEDİĞİM HALDE SONUCU 44 GOSTERIYOR? 
//   it("click button", () => {
    
//     const { wrapper, fnc } = setup(0);
    
//     const input1 = wrapper.find("input");
//     input1.simulate('change', {target: {value: '22'}});

//     console.log('tg..input1:',input1.html());
//     const button1 = wrapper.find("button");
//     button1.simulate("click");

//     expect(fnc.mock.calls.length).toBe(1);

    
//     const result = wrapper.find("label").text();
//     console.log('tg..result:', result)
//     expect(result).toBe("44");
//   });

// });


// it("should title-2 is true", () => {
//   const wrapper = shallow(<Double initialNum={32} />);

//   expect(wrapper.find("h5.title2").text()).toBe(" unit test title2")
//   expect(wrapper.find("h5.title1").text()).toEqual(" unit test title1")
// })  


// it("renders with shallow correctly", () => {
//   const wrapper = shallow(<Double initialNum={33} />)
//   expect(wrapper).toMatchSnapshot();
// });

// //? BEN BURADA CLICK EVENTINI SIMULE ETMEDIGIM HALDE LABEL.text'de inputun 2 katı geliyor. 
// it("label text gives correct value ", () => {
//   const wrapper = mount(<Double initialNum={44} />)
//   const text = wrapper.find("label").text(); //Enzyme ile bir elemente ulaşmak.
//   console.log("text:", text)
//   expect(text).toBe("88");
// });