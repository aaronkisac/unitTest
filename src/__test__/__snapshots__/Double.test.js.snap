// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Double component Cases renders with render correctly 1`] = `
<div
  className="page"
  id="divDouble"
>
  <h5
    className="title1"
  >
     unit test title1
  </h5>
  <h5
    className="title2"
  >
     unit test title2
  </h5>
  <div
    className="error"
  />
  <input
    defaultValue={33}
    id="number"
    onChange={[Function]}
    type="text"
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    Calculate
  </button>
  <div>
    <label>
      66
    </label>
  </div>
</div>
`;

exports[`Double component Cases renders with shallow correctly 1`] = `
<div
  className="page"
  id="divDouble"
>
  <h5
    className="title1"
  >
     unit test title1
  </h5>
  <h5
    className="title2"
  >
     unit test title2
  </h5>
  <div
    className="error"
  />
  <input
    defaultValue={33}
    id="number"
    onChange={[Function]}
    type="text"
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    Calculate
  </button>
  <div>
    <label>
      66
    </label>
  </div>
</div>
`;

exports[`renders with shallow correctly 1`] = `
<div
  className="page"
  id="divDouble"
>
  <h5
    className="title1"
  >
     unit test title1
  </h5>
  <h5
    className="title2"
  >
     unit test title2
  </h5>
  <div
    className="error"
  />
  <input
    defaultValue={33}
    id="number"
    onChange={[Function]}
    type="text"
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    Calculate
  </button>
  <div>
    <label>
      66
    </label>
  </div>
</div>
`;
